### DISCUSSION TITLE
Apply for GitHub issue forms

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2477] [Ideas]
## Info
GitHub currently has issue-forms running in a closed alpha.
Issue-forms for those who don't know are essentially enhanced Issue templates, allowing more freedom in designing issue templates.

I suggest that this repository should be signed up for the GitHub Issue-forms, which can be done through this Google Form: https://forms.gle/8PtmDALhWLvVqQrFA

There is a page providing some info about the issue forms: https://gh-community.github.io/issue-template-feedback/structured/

## Why apply?
Issue Forms have a lot of major benefits that this repository could benefit from:

- Required fields
  Will make the user require to fill out the field, select an option or similar before they can submit an issue.
- Better design
  Unlike the issue templates right now are issue forms not just the content of an MD file pasted into the text body. Instead will the selected template result in a form where you can select stuff or enter certain info into large or small text fields.
- Always the same structure
  The content created by the issue form will always be in the same structure. Not filled out fields will have `*No response*` as a replacement set.
- Easier filling out
  There are a few QOL changes compared to templates. As an example will you not need to change `[ ]` to `[x]` in checkboxes but can instead just click the checkbox itself to (un)tick it.
- Comments
  You can have simple `markdown` areas which are displayed as text in the form. This is useful to provide info instead of using the HTML comment tag.
  Additionally, can you also add descriptions to the different options for some explanation. Those descriptions won't be rendered/shown in the resulting issue.

However, are they also a few downsides to the issue forms:

- There is currently no "code block" type
  This means you can't just tell the user to put the text into that field and it would be turned into a code block. They have to do it themself. You can however pre-fill the field to make this a bit easier.
- No mobile support (yet)
  From what I gathered does GitHub's Mobile app not support issue forms yet, since it's still in alpha. So people submitting issues through the app would have either no Issue template or something broken/weird.
- No UI
  There is currently no UI for the easy creation of an issue form. This means you have to create the YAML files manually. There is also no live preview while editing the files.

## Image
Here's an image of one of my Issue templates that I use right now:
![image](https://user-images.githubusercontent.com/11576465/111863682-5969c580-895d-11eb-9e43-d6148fbd2ab7.png)

**Comment by squidfunk** [comment]
Thanks for suggesting! Looks interesting. However, given the many facets of issues I've seen in the past, I really would like to see this in production for some time before we switch to it. Also, things like screenshots and videos need to be supported.

**Comment by Andre601** [comment]
Issue forms are in public beta now, if it is of any interest for you.

Blog Post: https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/

**Reply by facelessuser** [reply]
Very cool. This is a much better approach than giving a Markdown template that people ignore and post their vague description under 🙃.

**Reply by squidfunk** [reply]
Cool! Would you like to submit a PR for a proposal how a form could look?

**Reply by Andre601** [reply]
> Cool! Would you like to submit a PR for a proposal how a form could look?

Done!
- #2790
